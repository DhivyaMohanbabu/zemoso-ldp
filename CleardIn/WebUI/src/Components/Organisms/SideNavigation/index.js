/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import {
  Icon, List, ListItem, ListItemIcon,
} from '@material-ui/core';
import PollOutlinedIcon from '@material-ui/icons/PollOutlined';
import AccessTimeIcon from '@material-ui/icons/AccessTime';
import SettingsIcon from '@material-ui/icons/Settings';
import NavIcon from '../../Atoms/SideNavigationIcon';
import excel from '../../../../assets/images/excel.png';

const useStyles = makeStyles((theme) => ({
  root: {
    width: '4rem',
    height: '100%',
  },
  list: {
    margin: '20px 0',
    justifyContent: 'center',
  },
  list2: {
    paddingTop: '22px',
    paddingBottom: '15px',
  },
  selctedIcon: {
    backgroundColor: theme.palette.clearedIn.selectedElementBGColor,
    /* borderWidth: '1px',
    borderStyle: 'solid',
    border: 'none',
    borderLeft: theme.palette.primary, */
    borderLeft: '1px solid blue',
  },
  listBox: {
    justifyContent: 'center',
    justifyItems: 'center',
  },
  listItem: {
    justifyContent: 'center',
  },
  listIcon: {
    width: '1.5rem',
    height: '1.5rem',
  },
  listIconRoot: {
    minWidth: 0,
  },
}));

function SideNavigation() {
  const classes = useStyles();
  const flag = true;
  return (
    <div className={classes.root}>
      <List data-testid="list" className={classes.listBox}>
        <ListItem className={classes.listItem}>
          <ListItemIcon
            classes={{
              root: classes.listIconRoot,
            }}
            className={classes.listIcon}
          >
            <Icon color="primary">io</Icon>
          </ListItemIcon>
        </ListItem>
        <ListItem className={[classes.listItem, classes.list]}>
          <ListItemIcon
            classes={{
              root: classes.listIconRoot,
            }}
            className={classes.listIcon}
          >
            <NavIcon icon={AccessTimeIcon} size="small" />
          </ListItemIcon>
        </ListItem>
        <div className={classes.selctedIcon}>
          <ListItem className={[classes.listItem, classes.list2]}>
            <ListItemIcon
              className={classes.listIcon}
              classes={{
                root: classes.listIconRoot,
              }}
            >
              <NavIcon icon={PollOutlinedIcon} size="small" isSelected={flag} />
            </ListItemIcon>
          </ListItem>
        </div>
        <ListItem className={[classes.listItem, classes.list]}>
          <ListItemIcon
            className={classes.listIcon}
            classes={{
              root: classes.listIconRoot,
            }}
          >
            <NavIcon icon={SettingsIcon} size="small" />
          </ListItemIcon>
        </ListItem>
        <ListItem className={[classes.listItem, classes.list]}>
          <ListItemIcon
            className={classes.listIcon}
            classes={{
              root: classes.listIconRoot,
            }}
          >
            <img src={excel} alt="close" />
          </ListItemIcon>
        </ListItem>
      </List>
    </div>
  );
}

export default SideNavigation;
