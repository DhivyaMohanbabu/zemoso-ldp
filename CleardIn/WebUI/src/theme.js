import { grey } from '@material-ui/core/colors';
import { createMuiTheme } from '@material-ui/core/styles';
// import 'typeface-roboto';

const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#1665d8',
      light: '#b0f1f0',
    },
    secondary: {
      main: '#9ea0a5',
      dark: grey,
    },
    error: {
      main: '#ffada5',

    },
    success: {
      main: '#daefa9',
    },
    info: {
      main: grey[900],
      light: '#fdd570',
      dark: grey[600],
    },
    text: {
      primary: '#1c1c21', // '#3e3f42',
      secondary: '#5c5c5c', // #6b6c6f',
    },
    grey: {
      100: '#f8fafb',
      200: '#f2f4f6',
      300: '#e8ecef',
      400: '#98a9bc',
      500: '#778ca2',
    },
    clearedIn: {
      grey: {
        100: '#f6f6f6',
        200: '#eeeeee',
        300: '#cccccc',
        400: '#949494',
        500: '#666666',
      },
      selectedElementBGColor: '#eaf0ff', // rgba(77, 124, 254, 0.12),
      dark: '#5c5c5c',
      bodyBackground: '#fafafa',
      blue: '#1d4cd7',
    },
    relationShip: {
      vendor: '#daefa9',
      customer: '#fdd570',
      partner: '#b0f1f0',
    },
  },
  typography: {
    fontFamily: ['Roboto', 'Rubik', 'sans-serif'].join(','),
    h1: {
      fontSize: '1.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h2: {
      fontSize: '1.625rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h3: {
      fontSize: '1.25rem',
      fontWeight: 300,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h4: {
      fontSize: '1.125rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h5: {
      fontSize: '1rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    h6: {
      fontSize: '1.25rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 1.05,
      letterSpacing: 'normal',
    },
    subtitle1: {
      fontSize: '0.938rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    subtitle2: {
      fontSize: '0.938rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    caption: {
      fontSize: '0.875rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    link: {
      fontSize: '0.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    body1: {
      fontSize: '0.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.14,
      letterSpacing: 'normal',
    },
    body2: {
      fontFamily: 'Rubik',
      fontSize: '0.875rem',
      fontWeight: 300,
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    button: {
      fontSize: '0.75rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
      textTransform: 'none',
    },
  },
  overrides: {
    MuiButton: {
      root: {
        maxWidth: '200px',
        display: 'block',
      },
      label: {
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
      },
    },
  },
});

export default theme;
