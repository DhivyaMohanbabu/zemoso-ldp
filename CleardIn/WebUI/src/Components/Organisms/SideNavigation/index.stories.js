import React from 'react';
import SideNavigation from '.';

export default {
  title: 'Organism/SideNavigation',
};

const Template = () => <SideNavigation />;

export const Default = Template.bind({});
