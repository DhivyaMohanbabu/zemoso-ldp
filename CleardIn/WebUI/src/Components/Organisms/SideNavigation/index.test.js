import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/core/styles';
import SideNavigation from '.';
import theme from '../../../theme';

describe('TableHeader', () => {
  it('should render new Row ', () => {
    const container = render(
      <ThemeProvider theme={theme}>
        <SideNavigation />
      </ThemeProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render TableHeader', () => {
    const { queryByTestId } = render(
      <ThemeProvider theme={theme}>
        <SideNavigation />
      </ThemeProvider>,
    );
    const component = queryByTestId('list');
    expect(component).not.toBeNull();
  });
});
