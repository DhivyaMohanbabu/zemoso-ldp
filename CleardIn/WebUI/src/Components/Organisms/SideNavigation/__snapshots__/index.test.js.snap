// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader should render new Row  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <ul
          class="MuiList-root makeStyles-listBox-5 MuiList-padding"
          data-testid="list"
        >
          <li
            class="MuiListItem-root makeStyles-listItem-6 MuiListItem-gutters"
          >
            <div
              class="MuiListItemIcon-root makeStyles-listIconRoot-8 makeStyles-listIcon-7"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root MuiIcon-colorPrimary"
              >
                io
              </span>
            </div>
          </li>
          <li
            class="MuiListItem-root makeStyles-listItem-6 makeStyles-list-2 MuiListItem-gutters"
          >
            <div
              class="MuiListItemIcon-root makeStyles-listIconRoot-8 makeStyles-listIcon-7"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-root makeStyles-notSelected-10 MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
                <path
                  d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
                />
              </svg>
            </div>
          </li>
          <div
            class="makeStyles-selctedIcon-4"
          >
            <li
              class="MuiListItem-root makeStyles-listItem-6 makeStyles-list2-3 MuiListItem-gutters"
            >
              <div
                class="MuiListItemIcon-root makeStyles-listIconRoot-8 makeStyles-listIcon-7"
              >
                <div
                  class="MuiGrid-root makeStyles-root-9"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V5h14v14zM7 10h2v7H7zm4-3h2v10h-2zm4 6h2v4h-2z"
                    />
                  </svg>
                </div>
              </div>
            </li>
          </div>
          <li
            class="MuiListItem-root makeStyles-listItem-6 makeStyles-list-2 MuiListItem-gutters"
          >
            <div
              class="MuiListItemIcon-root makeStyles-listIconRoot-8 makeStyles-listIcon-7"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-root makeStyles-notSelected-10 MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
                />
              </svg>
            </div>
          </li>
          <li
            class="MuiListItem-root makeStyles-listItem-6 makeStyles-list-2 MuiListItem-gutters"
          >
            <div
              class="MuiListItemIcon-root makeStyles-listIconRoot-8 makeStyles-listIcon-7"
            >
              <img
                alt="close"
                src="test-file-stub"
              />
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <ul
        class="MuiList-root makeStyles-listBox-5 MuiList-padding"
        data-testid="list"
      >
        <li
          class="MuiListItem-root makeStyles-listItem-6 MuiListItem-gutters"
        >
          <div
            class="MuiListItemIcon-root makeStyles-listIconRoot-8 makeStyles-listIcon-7"
          >
            <span
              aria-hidden="true"
              class="material-icons MuiIcon-root MuiIcon-colorPrimary"
            >
              io
            </span>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-listItem-6 makeStyles-list-2 MuiListItem-gutters"
        >
          <div
            class="MuiListItemIcon-root makeStyles-listIconRoot-8 makeStyles-listIcon-7"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-root makeStyles-notSelected-10 MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
              <path
                d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
              />
            </svg>
          </div>
        </li>
        <div
          class="makeStyles-selctedIcon-4"
        >
          <li
            class="MuiListItem-root makeStyles-listItem-6 makeStyles-list2-3 MuiListItem-gutters"
          >
            <div
              class="MuiListItemIcon-root makeStyles-listIconRoot-8 makeStyles-listIcon-7"
            >
              <div
                class="MuiGrid-root makeStyles-root-9"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V5h14v14zM7 10h2v7H7zm4-3h2v10h-2zm4 6h2v4h-2z"
                  />
                </svg>
              </div>
            </div>
          </li>
        </div>
        <li
          class="MuiListItem-root makeStyles-listItem-6 makeStyles-list-2 MuiListItem-gutters"
        >
          <div
            class="MuiListItemIcon-root makeStyles-listIconRoot-8 makeStyles-listIcon-7"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-root makeStyles-notSelected-10 MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
              />
            </svg>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-listItem-6 makeStyles-list-2 MuiListItem-gutters"
        >
          <div
            class="MuiListItemIcon-root makeStyles-listIconRoot-8 makeStyles-listIcon-7"
          >
            <img
              alt="close"
              src="test-file-stub"
            />
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
